pkg/test/collector_perf_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !race
+
+package test
+
+import (
+	"flag"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"github.com/vmware/go-ipfix/pkg/collector"
+	"github.com/vmware/go-ipfix/pkg/exporter"
+	"github.com/vmware/go-ipfix/pkg/registry"
+)
+
+func init() {
+	// Load the global registry
+	registry.LoadRegistry()
+}
+
+/*
+    Sample output:
+	go test -test.v -run=Benchmark -test.benchmem -bench=BenchmarkMultipleExportersToCollector -memprofile memprofile.out -cpuprofile profile.out
+
+	goos: darwin
+	goarch: amd64
+	pkg: github.com/vmware/go-ipfix/pkg/test
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkMultipleExportersToCollector
+	BenchmarkMultipleExportersToCollector-12              12          88131538 ns/op         1460866 B/op      42195 allocs/op
+	PASS
+	ok      github.com/vmware/go-ipfix/pkg/test     7.233s
+*/
+func BenchmarkMultipleExportersToCollector(b *testing.B) {
+	disableLogToStderr()
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+	collectorInput := collector.CollectorInput{
+		Address:       "127.0.0.1:0",
+		Protocol:      "tcp",
+		MaxBufferSize: 1024,
+		TemplateTTL:   0,
+	}
+	cp, err := collector.InitCollectingProcess(collectorInput)
+	if err != nil {
+		b.Fatalf("cannot start collecting process on %s: %v", cp.GetAddress().String(), err)
+	}
+	go cp.Start()
+	waitForCollectorStatus(b, cp, true)
+	b.ResetTimer()
+	for i := 0; i < 1000; i++ {
+		exporterInput := exporter.ExporterInput{
+			CollectorAddress:    cp.GetAddress().String(),
+			CollectorProtocol:   cp.GetAddress().Network(),
+			ObservationDomainID: uint32(i),
+		}
+		exporter, err := exporter.InitExportingProcess(exporterInput)
+		if err != nil {
+			b.Errorf("cannot start exporting process: %v", err)
+		}
+		templateID := exporter.NewTemplateID()
+		exporter.SendSet(createTemplateSet(templateID, false))
+		exporter.SendSet(createDataSet(templateID, true, false, true))
+	}
+	count := 0
+	for range cp.GetMsgChan() {
+		count++
+		if count == 2000 {
```

Comment:
We can configure this number. Make each exporter send 10K records?

Reply:
Sure. When I make each exporter send 10K records, found that `Message length 95 is larger than size read from buffer 45` you have previously mentioned. I feel it should be related to the Logstash tcp decoding issue. Working on triaging it.